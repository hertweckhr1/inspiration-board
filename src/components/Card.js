import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';
// WHY IS MY POST FROM POSTMAN NOT WORKING? SI SE PUEDE
const Card = (props) => {
    const text = props.text;
    const id = props.id;
    let icon = props.emoji
    if (icon !== null) {
      icon = emoji.getUnicode(icon)
    }
    // const emoji = emoji.getUnicode(props.emoji);
    return (
      <div className="card card__content">
        <p className="card__content-text">{text}</p>
        <p className="card__content-emoji">{icon}</p>
        <button
        className="card__delete"
        onClick={() => {props.deleteCardCallback(props.id)}}
        >Delete Note</button>

      </div>
    )
  }

Card.propTypes = {
  text: PropTypes.string,
  emoji: PropTypes.string,
  id: PropTypes.number,
};

export default Card;
