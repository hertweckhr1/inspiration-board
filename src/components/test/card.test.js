import React from 'react';
import Card from '../Card';
import { shallow } from 'enzyme';

describe('card', () => {
  it('will match the card snapshot', () => {
    const wrapper = shallow( <Card key={1}
            id={1}
            text="Woo hoo!"
            emoji="heart_eyes"
            deleteCardCallback={() => {}} />)

    expect(wrapper).toMatchSnapshot();
  });
})
