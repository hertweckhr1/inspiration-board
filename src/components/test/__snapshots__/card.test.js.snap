// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card will match the card snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    deleteCardCallback={[Function]}
    emoji="heart_eyes"
    id={1}
    text="Woo hoo!"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="card__content-text"
        >
          Woo hoo!
        </p>,
        <p
          className="card__content-emoji"
        >
          üòç
        </p>,
        <button
          className="card__delete"
          onClick={[Function]}
        >
          Delete Note
        </button>,
      ],
      "className": "card card__content",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Woo hoo!",
          "className": "card__content-text",
        },
        "ref": null,
        "rendered": "Woo hoo!",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "üòç",
          "className": "card__content-emoji",
        },
        "ref": null,
        "rendered": "üòç",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Delete Note",
          "className": "card__delete",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Delete Note",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="card__content-text"
          >
            Woo hoo!
          </p>,
          <p
            className="card__content-emoji"
          >
            üòç
          </p>,
          <button
            className="card__delete"
            onClick={[Function]}
          >
            Delete Note
          </button>,
        ],
        "className": "card card__content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Woo hoo!",
            "className": "card__content-text",
          },
          "ref": null,
          "rendered": "Woo hoo!",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "üòç",
            "className": "card__content-emoji",
          },
          "ref": null,
          "rendered": "üòç",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Delete Note",
            "className": "card__delete",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Delete Note",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
